### PR TITLE
A collection of documentation updates

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] Create a news fragment with `towncrier create <IssueNumber>.<type>` which will be
  included in the changelog. `<type>` can be one of: new_check, removed_check, extension,
  false_positive, false_negative, bugfix, other, internal. If necessary you can write
  details or offer examples on how the new change is supposed to work.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Closes https://github.com/PyCQA/pylint/issues/7039, Closes https://github.com/PyCQA/pylint/issues/6900, Refs https://github.com/PyCQA/pylint/issues/7478

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -66,6 +66,16 @@ ecosystem of existing plugins for popular frameworks such as `pylint-django`_ or
...
+.. note::
+
+    Pylint supports the Python standard library out of the box. Third-party
+    libraries are not always supported, so a plugin might be needed. A good place
+    to start is ``PyPI`` which often returns a plugin by searching for
+    ``pylint <library>``. `pylint-django`_ and `pylint-pydantic`_ are examples
+    of such plugins.
+
+.. _`pylint-pydantic`: https://pypi.org/project/pylint-pydantic
+
```
Comment (mbyrnepr2) [inline README.rst:78]: Perhaps we should also link to plugin usage so that a user understands how to use the plugins.

File: README.rst [hunk 2]
```diff
@@ -66,6 +66,16 @@ ecosystem of existing plugins for popular frameworks such as `pylint-django`_ or
 .. _`pylint-django`: https://github.com/PyCQA/pylint-django
 .. _`pylint-sonarjson`: https://github.com/omegacen/pylint-sonarjson
 
+.. note::
```
Comment (mbyrnepr2) [inline README.rst:69]:
```suggestion
.. Plugins::
```

Or something similarly specific.

**Review by mbyrnepr2 — COMMENTED** [review]
Thanks for doing this!

File: README.rst (hunk 1 above)
Comment (mbyrnepr2) [reply README.rst:78]: The Readme also contains plugin information in the preceding paragraph; lets put them together?

File: doc/user_guide/configuration/index.rst [hunk 3]
```diff
@@ -18,6 +18,13 @@ For example::
 In practice, it is often better to create a minimal configuration file which only contains
 configuration overrides. For all other options, Pylint will use its default values.
 
+.. note::
+
+    The internals that create the configuration files fall back to the default values if
+    not other value was given. This means that some values depend on the interpreter that
```
Comment (mbyrnepr2) [inline doc/user_guide/configuration/index.rst:24]:
```suggestion
    no other value was given. This means that some values depend on the interpreter that
```

File: README.rst (hunk 1 above)
Comment (DanielNoord) [reply README.rst:78]: Yeah I wasn't sure how to combine those. Shall we make one `note`? Or one main paragraph?

Comment (mbyrnepr2) [reply README.rst:78]: I was thinking of converting your `note` paragraph to be `plugins` then add the existing plugin info here + your new content. Happy to go another way if you prefer also.

File: README.rst (hunk 2 above)
Comment (DanielNoord) [reply README.rst:69]: See:
https://pylint--7512.org.readthedocs.build/en/7512/

Comment (mbyrnepr2) [reply README.rst:69]: Oh sorry I may have misunderstood. I thought a paragraph named `plugins` would be more on-point but whichever way you prefer.

Comment (mbyrnepr2) [reply README.rst:69]: Ah ok I see the different rendering for note. I think the main thing is to have all plugin-related info in the same note or paragraph etc. just to keep them close together.

**Review by mbyrnepr2 — APPROVED** [review]

File: README.rst [hunk 4]
```diff
@@ -59,10 +59,19 @@ will know that ``argparse.error(...)`` is in fact a logging call and not an argp
...
-`pylint-sonarjson`_.
+own checks (for example, for internal libraries or an internal rule). Pylint also has an
+ecosystem of existing plugins for popular frameworks and third party libraries.
 
+.. note::
+
+    Pylint supports the Python standard library out of the box. Third-party
+    libraries are not always supported, so a plugin might be needed. A good place
+    to start is ``PyPI`` which often returns a plugin by searching for
+    ``pylint <library>``. `pylint-django`_ and `pylint-pydantic`_, `pylint-django`_ and
```
Comment (DanielNoord) [inline README.rst:70]:
```suggestion
    ``pylint <library>``. `pylint-pydantic`_, `pylint-django`_ and
```

**Review by mbyrnepr2 — DISMISSED** [review]

File: doc/data/messages/e/exec-used/details.rst [hunk 5]
```diff
@@ -1 +1,10 @@
...
+The available methods and variables used in ``exec()`` may introduce a security hole.
+You can restrict the use of these variables and methods by passing optional globals
+and locals parameters (dictionaries) to the ``exec()`` method.
+
+However, use of ``exec`` is still insecure. For example, consider the following call
+that writes a file to the user's system:
+
+.. code-block:: python
+
+    exec("""\nwith open("file.txt", "w", encoding="utf-8") as file:\n file.write("test")\n""")
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/e/exec-used/details.rst:10]:
```suggestion
    exec("""\nwith open("file.txt", "w", encoding="utf-8") as file:\n file.write("# code as nefarious as imaginable")\n""")
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by mbyrnepr2 — APPROVED** [review]

**Review by Gornoka — COMMENTED** [review]
thanks for adding the hint regarding `py-version`

**Pull request merged**